app/src/main/java/com/spendster/presentation/WelcomeActivity.java

```diff
@@ -0,0 +1,130 @@
+    private ArrayList<Fragment> slideFragments;
+    private ImageView[] dots;
+    private View dotsLayout;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_welcome);
+
+        initWelcomeActivity();
+        initViewPager();
+
+        btnSkip.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                launchThirdScreen();
+            }
+        });
+
+        btnNext.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                int current = getItem(1);
+                if (current < slideFragments.size()) {
+                    viewPager.setCurrentItem(current);
+                } else {
+                    launchThirdScreen();
+                }
+            }
+        });
+    }
+
+    private void initViewPager() {
+        ViewPagerAdapter viewPagerAdapter = new ViewPagerAdapter(getSupportFragmentManager(), slideFragments);
+        viewPager.setAdapter(viewPagerAdapter);
+        viewPager.addOnPageChangeListener(viewPagerPageChangeListener);
+    }
+
+    private void initWelcomeActivity() {
+        viewPager = findViewById(R.id.view_pager);
+        btnSkip = findViewById(R.id.btnSkip);
+        btnNext = findViewById(R.id.btnNext);
+        dotsLayout = findViewById(R.id.bottom_nav_dots);
+        btnSignUp = findViewById(R.id.btnSignUp);
+        btnLogin = findViewById(R.id.btnLogin);
+        initSlideFragments();
+        initBottomNavDots();
+        selectActivePageDot(0);
+    }
+
+    private void initSlideFragments(){
+        slideFragments = new ArrayList<>();
+        slideFragments.add(SlideFragment.newInstance(R.layout.slide1));
+        slideFragments.add(SlideFragment.newInstance(R.layout.slide2));
+        slideFragments.add(SlideFragment.newInstance(R.layout.slide3));
+    }
+
+    private void initBottomNavDots() {
+        dots = new ImageView[slideFragments.size()];
+        dots[0] = dotsLayout.findViewById(R.id.bottom_nav_dot_1);
+        dots[1] = dotsLayout.findViewById(R.id.bottom_nav_dot_2);
+        dots[2] = dotsLayout.findViewById(R.id.bottom_nav_dot_3);
+    }
+
+    private void selectActivePageDot(int currentPage) {
+        for (int i = 0; i < slideFragments.size(); i++){
+            if(i == currentPage){
+                dots[i].setImageDrawable(ContextCompat.getDrawable(this, R.drawable.selected_dot));
+            }
+            else{
+                dots[i].setImageDrawable(ContextCompat.getDrawable(this, R.drawable.default_dot));
+            }
+        }
+    }
+
+    ViewPager.OnPageChangeListener viewPagerPageChangeListener = new ViewPager.OnPageChangeListener() {
+
+        @Override
+        public void onPageSelected(int position) {
+            selectActivePageDot(position);
+            if (position == slideFragments.size() - 1) {
```

Comment:
@SimonMaha extract this into a method like that:
if(isLastSlide(position))

It will be much more readable